packages/core/rollup/styleInject.ejs

```diff
@@ -1,21 +1,25 @@
 function styleInject(css) {
-    if(typeof document === 'undefined') return;
-    const head = document.head || document.getElementsByTagName('head')[0];
     const id = "<%= hashValue %>_<%= version %>";
-    const styleExists = head.querySelector("#" + id);
-
-    if(styleExists) return;
-
-    const style = document.createElement('style');
-    style.id = id;
-
-    if (head.firstChild) {
-        head.insertBefore(style, head.firstChild);
-    } else {
-        head.appendChild(style);
+    if(typeof document !== 'undefined') {
+        const head = document.head || document.getElementsByTagName('head')[0];
+        const styleExists = head.querySelector("#" + id);
+    
+        if(styleExists) return;
+    
+        const style = document.createElement('style');
+        style.id = id;
+    
+        if (head.firstChild) {
+            head.insertBefore(style, head.firstChild);
+        } else {
+            head.appendChild(style);
+        }
+    
+        style.appendChild(document.createTextNode(css));
+    } else if (typeof globalThis !== 'undefined' && globalThis.injectedStyles) {
+        // support SSR environments
+        globalThis.injectedStyles[id] = css;
```

Comment:
this is not a direct dependency of vibe, should it be? (it's not used anywhere in the code, only in your addition so might be good to add it as a direct dependency?)